Assignment/test.c

```diff
@@ -56,20 +51,40 @@ void DecToHexaTest () {
       "7FFFFFFF",
       "80000001" };
    int testCount = sizeof (input2) / sizeof (input2[0]);
-   char actualhexa[35];
    int terminalWidth = 100;
    printf ("---Decimal to Hexa---\n\n");
    for (int i = 0; i < testCount; i++) {
-      DecToHexa (input2[i], actualhexa);
+      char* actualhexa = DecToHexa (input2[i]);
       printf ("Input %12d => ", input2[i]);
-      // Check result and print at the rightmost corner
-      if (strcmp (actualhexa, expected_output[i]) == 0) PrintRightAligned (GREEN_TEXT "Pass" RESET_TEXT, terminalWidth);
-      else PrintRightAligned (RED_TEXT "Fail" RESET_TEXT, terminalWidth);
+      PrintResult ((strcmp (actualhexa, expected_output[i]) == 0) ? GREEN_TEXT "Pass" RESET_TEXT : RED_TEXT "Fail" RESET_TEXT, terminalWidth);
    }
 }
 
+void UserInput () {
+   int number;
+   printf ("Enter a decimal number: ");
+   scanf_s ("%d", &number);
+   printf ("\nBinary: %s\n", DecToBinary (number));
+   printf ("Hexadecimal: %s\n", DecToHexa (number));
+}
+
 int main () {
-   DecToBinaryTest ();
-   DecToHexaTest ();
+   int choice;
+   printf ("1. Test Cases\n");
+   printf ("2. User Input\n");
+   printf ("Enter your choice (1 or 2): ");

```

Comment:
You can combine statements wherever possible and write it as a single statement. This can fewer the system calls and improve performance.
printf ("1. Test Cases\n 2. User Input\n Enter your choices (1 or 2): ");